plugins/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/FirKotlinUastResolveProviderService.kt

```diff
@@ -29,4 +31,8 @@ interface FirKotlinUastResolveProviderService : BaseKotlinUastResolveProviderSer
                 return null
         }
     }
+
+    override fun getExpressionType(uExpression: UExpression): PsiType? {
```

Comment:
FYI, the future changes regarding this API are here: https://github.com/JetBrains/kotlin/pull/4429